src/qinfer/derived_models.py

```diff
@@ -380,6 +381,128 @@ def fisher_information(self, modelparams, expparams):
         )
         return two_outcome_fi * expparams['n_meas']
 
+class GaussianHyperparameterizedModel(DerivedModel):
+    """
+    Model representing a two-outcome model viewed through samples
+    from one of two distinct Gaussian distributions. This model adds four new
+    model parameters to its underlying model, respectively representing the
+    mean outcome conditioned on an underlying 0, the mean outcome conditioned
+    on an underlying 1, and the variance of outcomes conditioned in each case.
+
+    :param qinfer.abstract_model.Model underlying_model: An instance of a two-
+        outcome model to be viewed through Gaussian distributions.
+    """
+
+    def __init__(self, underlying_model):
+        super(GaussianHyperparameterizedModel, self).__init__(underlying_model)
+
+        if not (underlying_model.is_n_outcomes_constant and underlying_model.n_outcomes(None) == 2):
+            raise ValueError("Decorated model must be a two-outcome model.")
+
+    ## PROPERTIES ##
+
+    @property
+    def decorated_model(self):
+        # Provided for backcompat only.
+        return self.underlying_model
+
+    @property
+    def modelparam_names(self):
+        return self.underlying_model.modelparam_names + [
+            r'\mu_0', r'\mu_1',
+            r'\sigma_0^2', r'\sigma_1^2'
+        ]
+    
+    @property
+    def n_modelparams(self):
+        return len(self.modelparam_names)
+    
+    ## METHODS ##
+    
+    def domain(self, expparams):
+        return [RealDomain()] * len(expparams)
+    
+    def are_expparam_dtypes_consistent(self, expparams):
+        return True
+
+    def are_models_valid(self, modelparams):
+        orig_mps = modelparams[:, :-4]
+        sigma2 = modelparams[:, -2:]
+        
+        return np.all([
+            self.underlying_model.are_models_valid(orig_mps),
+            np.all(sigma2 > 0, axis=-1)
+        ], axis=0)
+
+    def underlying_likelihood(self, binary_outcomes, modelparams, expparams):
+        """
+        Given outcomes hypothesized for the underlying model, returns the likelihood
+        which which those outcomes occur.
+        """
+        original_mps = modelparams[..., :-4]
+        return self.underlying_model.likelihood(binary_outcomes, original_mps, expparams)
+
+    def likelihood(self, outcomes, modelparams, expparams):
+        # By calling the superclass implementation, we can consolidate
+        # call counting there.
+        super(GaussianHyperparameterizedModel, self).likelihood(outcomes, modelparams, expparams)
+
+        # We want these to broadcast to the shape
+        #     (idx_underlying_outcome, idx_outcome, idx_modelparam, idx_experiment).
+        # Thus, we need shape
+        #     (idx_underlying_outcome,           1, idx_modelparam,              1).
+        mu = (modelparams[:, -4:-2].T)[:, None, :, None]
```

Comment:
``np.newaxis`` is used a bit more consistently elsewhere, I think?

Reply:
Good point, I was being a bit sloppy here, but `np.newaxis` is much easier to read. I'll fix that, then.